### PR TITLE
Remove 2FA Restrictions for AAD and Certificate management

### PR DESCRIPTION
**Pull request by shishirx34**
Addresses #6005 

Both restrictions removed for User/Org certificate management.

image

Manage packages, AAD account without 2FA:

image

MSA account without 2FA:
image

### CODE REVIEW

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 1]
```diff
@@ -549,6 +543,13 @@ public virtual JsonResult GetCertificate(string accountName, string thumbprint)
             return Json(HttpStatusCode.OK, certificates, JsonRequestBehavior.AllowGet);
         }
 
+        private bool CanManageCertificates(User currentUser, User account)
+        {
+            var wasAADLoginOrMultiFactorAuthenticated = User.WasMultiFactorAuthenticated() || User.WasAzureActiveDirectoryAccountUsedForSignin();
+            return ActionsRequiringPermissions.ManageAccount
+                .CheckPermissions(currentUser, account) == PermissionsCheckResult.Allowed && wasAADLoginOrMultiFactorAuthenticated;
```
Comment (chenriksson) [inline src/NuGetGallery/Controllers/AccountsController.cs]: Don't need permissions check if 2FA or AAD checks failed. Suggest refactoring to return failure fast.

File: src/NuGetGallery/Extensions/PrincipalExtensions.cs [hunk 2]
```diff
@@ -218,14 +218,19 @@ public static bool TryRemoveClaim(this IIdentity identity, string claimType)
         /// </summary>
         /// <param name="identity">IIdentity from which the claim is to be removed</param>
         /// <param name="claimType">The claim type to be added</param>
+        /// <param name="claimValue">The claim value for the type to be added, null will add default boolean claim</param>
         /// <returns>True if successfully able to add the claim, false otherwise</returns>
-        public static bool TryAddClaim(this IIdentity identity, string claimType)
+        public static bool TryAddClaim(this IIdentity identity, string claimType, string claimValue = null)
```
Comment (loic-sharma) [inline src/NuGetGallery/Extensions/PrincipalExtensions.cs]: >TryAddClaim [](start = 27, length = 11)

What's this `claimValue` parameter used for? I don't see anything using it

Comment (shishirx34) [reply src/NuGetGallery/Extensions/PrincipalExtensions.cs]: Tests use it, added for convenience: https://github.com/NuGet/NuGetGallery/pull/6017/files#diff-6d61bf27406c5d5566e287664b5c4f46R2038

**Comment by shishirx34** [comment]
@dtivel, @chenriksson  - Removed the restriction of 2FA for AAD for setting the required signers, see the screenshot in the description. Please review.

**Comment by chenriksson** [comment]
@shishirx34 LGTM

**Pull request merged**